src/rulesets/Base/array.jl

```diff
@@ -527,3 +527,42 @@ function _extrema_dims(x, dims)
     end
     return y, extrema_pullback_dims
 end
+
+#####
+##### `pmap`
+#####
+
+using UUIDs # of course we don't want this;
```

Comment:
Use `gensym` maybe?

Reply:
Hmm. I've modified the code to use a counter instead of a uuid, but might `gensymb` be an even better solution? Could it be used to create an array with a unique name on all of the worker processors, as opposed to having this array reside in `save_backs[key]` like it currently does? (I don't have any experience with meta-programming in Julia)

Reply:
Do you have any pointers on how I might use 'gensym' here? I've never worked with Julia meta-programming. 

(I've currently replaced uuid with a counter, but I just realized that this would fail in the case of nested 'pmap''s as different processors' counters would start conflicting in the dictionary :/)